core/dbt/parser/read_files.py

```diff
@@ -153,104 +172,243 @@ def generate_dbt_ignore_spec(project_root):
-        saved_files,
-        dbt_ignore_spec,
-    )
+        for parse_ft, file_type_info in file_types.items():
+            project_files[file_type_info["parser"]] = read_files_for_parser(
+                project,
+                self.files,
+                parse_ft,
+                file_type_info,
+                self.saved_files,
+                dbt_ignore_spec,
+            )
 
-    project_files["SingularTestParser"] = read_files_for_parser(
-        project,
-        files,
-        project.test_paths,
-        [".sql"],
-        ParseFileType.SingularTest,
-        saved_files,
-        dbt_ignore_spec,
-    )
 
-    # all generic tests within /tests must be nested under a /generic subfolder
-    project_files["GenericTestParser"] = read_files_for_parser(
-        project,
-        files,
-        project.generic_test_paths,
-        [".sql"],
-        ParseFileType.GenericTest,
-        saved_files,
-        dbt_ignore_spec,
-    )
+@dataclass
+class ReadFilesFromDiff:
+    root_project_name: str
+    all_projects: Mapping[str, Project]
+    file_diff: FileDiff
+    files: Dict[str, AnySourceFile] = field(default_factory=dict)
+    # Is saved_files required? Could a file diff contain the entire project?
+    saved_files: Dict[str, AnySourceFile] = field(default_factory=dict)
+    project_parser_files: Dict = field(default_factory=dict)
+    project_file_types: Dict = field(default_factory=dict)
 
-    project_files["SeedParser"] = read_files_for_parser(
-        project,
-        files,
-        project.seed_paths,
-        [".csv"],
-        ParseFileType.Seed,
-        saved_files,
-        dbt_ignore_spec,
-    )
+    def read_files(self):
+        # Copy the base file information from the existing manifest.
+        # We will do deletions, adds, changes from the file_diff to emulate
+        # a complete read of the project file system.
+        for file_id, source_file in self.saved_files.items():
+            if isinstance(source_file, SchemaSourceFile):
+                file_cls = SchemaSourceFile
+            else:
+                file_cls = SourceFile
+            new_source_file = file_cls(
+                path=source_file.path,
+                checksum=source_file.checksum,
+                project_name=source_file.project_name,
+                parse_file_type=source_file.parse_file_type,
+                contents=source_file.contents,
+            )
+            self.files[file_id] = new_source_file
 
-    project_files["DocumentationParser"] = read_files_for_parser(
-        project,
-        files,
-        project.docs_paths,
-        [".md"],
-        ParseFileType.Documentation,
-        saved_files,
-        dbt_ignore_spec,
-    )
+        # Now that we have a copy of the files, remove deleted files
+        # For now, we assume that all files are in the root_project, until
+        # we've determined whether project name will be provided or deduced
+        # from the directory.
```

Comment:
As it turns out, the only other local files besides the root project are local dependencies, and there currently is no way to connect up a local package with its directory, and in addition local dependency "files" are generated off of where they're copied by deps. This issue has not been resolved, because it's tied up in the question about what to do about deps.